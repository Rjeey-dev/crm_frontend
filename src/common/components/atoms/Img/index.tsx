import React, {CSSProperties, FormEvent} from 'react';
// @ts-ignore
import { LazyLoadImage } from 'react-lazy-load-image-component';

interface IProps {
    src: string,
    classes?: string,
    alt?: string,
    title?: string,
    width?: number,
    height?: number,
    onClick?: (event: FormEvent<HTMLImageElement>) => void,
    attributes?: () => void,
    style?: CSSProperties,
    lazyLoad?: boolean
}

function Img(props: IProps) {
    const lazyLoad = props.hasOwnProperty('lazyLoad') ? props.lazyLoad : true;
    const {classes, src, alt, title, width, height, onClick, style} = props;

    const options = {
        className: classes,
        src,
        alt,
        title,
        width,
        height,
        onClick,
        style
    };

    if (lazyLoad) {
        return <LazyLoadImage placeholderSrc='/static/images/preloader.gif' effect="blur" {...props.attributes} {...options}/>
    }

    return <img {...props.attributes} {...options}/>;
}

export default Img;
