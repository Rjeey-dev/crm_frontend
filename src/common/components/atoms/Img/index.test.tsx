import { shallow } from 'enzyme';
import React from 'react';

import Img from './index';

it("Renders <Img/> with correct properties", () => {
    const output = shallow(
        <Img classes='class' src='/images/image.png' alt='alt' title='title' width={30}/>
    ).html();

    expect(output).toMatchSnapshot();
});

it("Renders <Img/> with only required props", () => {
    const output = shallow(
        <Img src='/images/image.png'/>
    ).html();

    expect(output).toMatchSnapshot();
});