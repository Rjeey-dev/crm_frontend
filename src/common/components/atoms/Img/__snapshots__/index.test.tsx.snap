// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Img/> with correct properties 1`] = `"<img class=\\"class\\" src=\\"/images/image.png\\" alt=\\"alt\\" title=\\"title\\" width=\\"30\\"/>"`;

exports[`Renders <Img/> with only required props 1`] = `"<img src=\\"/images/image.png\\"/>"`;
